core/amber/src/main/protobuf/edu/uci/ics/amber/engine/architecture/sendsemantics/partitionings.proto

```diff
@@ -45,3 +46,7 @@ message RangeBasedShufflePartitioning {
   int64 rangeMax = 5;
 }
 
+message BroadcastPartitioning{
+  int32 batchSize = 1;
+  repeated edu.uci.ics.amber.engine.common.ActorVirtualIdentity receivers = 2;
```

Comment:
do we still need the receivers if it simply broadcasts to all? or does the frontend (not in this PR) intended to have user to choose which workers to broadcast to?

Reply:
it's fine to leave it here if the partitioner needs this information. but better to be explicit if `receivers` will be all workers of the downstream operator or not.

Reply:
But I think we still need to have at least know how many receivers?

Reply:
sure. currently, only the controller knows the receiver's names (ids), and the controller needs to find a way to tell a worker this information. So yeah you can keep the receivers in this partitioning message.